chapter12/index.md

```diff
@@ -48,39 +48,44 @@ Now let's `SELECT Person.strength;` and see if it works:
 
 Looks like it worked.
 
-So now let's overload the `fight()` function. Right now it only works for one `Person` vs. another `Person`, but in the book all the characters get together to try to defeat Dracula. We'll need to overload the function so that more than one character can work together to fight. There are a lot of ways to do it, but we'll choose a simple one:
+So now let's overload the `fight()` function. Right now it only works for one `Person` vs. another `Person`, but in the book all the characters get together to try to defeat Dracula. We'll need to overload the function so that more than one character can work together to fight.
 
 ```sdl
-function fight(names: str, one: int16, two: str) -> str
+function fight(people_names: array<str>, opponent: Person) -> str
   using (
-    WITH opponent := assert_single((SELECT Person FILTER .name = two))
+    WITH
+        people := (SELECT Person FILTER .name IN array_unpack(people_names)),
     SELECT
-        names ++ ' win!' IF one > opponent.strength ELSE
-        two ++ ' wins!'
+        (array_join(people_names, ', ') ?? 'Fighters') ++ ' win!'
+        IF (sum(people.strength) ?? 0) > (opponent.strength ?? 0)
+        ELSE (opponent.name ?? 'Opponent') ++ ' wins!'
```

Comment:
You don't need to coalesce the `sum` it already handles empty set correctly:
```
edgedb> select sum(<int64>{});
{0}
```

We also don't need to coalesce things like `array_join` because the function signature makes it required (by default), so it won't produce empty sets. However, handling the `[]` is an entirely different story and can be done with an `if`.

I think this is a cleaner version with fewer unnecessary parts:
```
WITH
    people := (SELECT Person FILTER contains(people_names, .name)),
    names := ['Weak fighters'] 
        IF people_names = <array<str>>[] 
        ELSE people_names,
SELECT
    array_join(names, ', ') ++ ' win!'
    IF sum(people.strength) > (opponent.strength ?? 0)
    ELSE (opponent.name ?? 'Opponent') ++ ' wins!'
```

Reply:
Actually, now that I think about it it may be unnecessary to handle `[]` of fighters because they can never win (`0 > 0` is still `false`). So it simply doesn't matter what we call them.
```
WITH
    people := (SELECT Person FILTER contains(people_names, .name)),
SELECT
    array_join(people_names, ', ') ++ ' win!'
    IF sum(people.strength) > (opponent.strength ?? 0)
    ELSE (opponent.name ?? 'Opponent') ++ ' wins!'
```

Reply:
Good suggestions on the the function. I'll get those implemented.

I tried running "That's a lot of hassle, and it's not something we're ready to do right now," through the Hemingway editor which estimates readability, and it tells me that sentence is a 4th grade reading level. My feeling is that it's pretty simple, but I agree that I can go simpler than "comprehend." I'll rework that.